pdns/webserver.cc

```diff
@@ -130,12 +130,11 @@ void WebServer::apiWrapper(WebServer::HandlerFunction handler, HttpRequest* req,
 
   resp->headers["access-control-allow-origin"] = "*";
 
-  if (d_apikey.empty()) {
-    g_log<<Logger::Error<<req->logprefix<<"HTTP API Request \"" << req->url.path << "\": Authentication failed, API Key missing in config" << endl;
-    throw HttpUnauthorizedException("X-API-Key");
-  }
+  bool auth_ok = false;
 
-  bool auth_ok = req->compareHeader("x-api-key", d_apikey) || req->getvars["api-key"] == d_apikey;
+  if (!d_apikey.empty()) {
+    auth_ok = req->compareHeader("x-api-key", d_apikey) || req->getvars["api-key"] == d_apikey;
+  }
 
```

Comment:
The logic is now: if the api-key is passed in but does not match, password auth is still tried if allowed. I think that can be confusing.